--- compiler/rustc_hir/src/hir.rs
+++ compiler/rustc_hir/src/hir.rs
@@ -4938,4 +4938,5 @@
 // Some nodes are used a lot. Make sure they don't unintentionally get bigger.
+#[cfg(not(rust_compiler="mrustc"))]
 #[cfg(target_pointer_width = "64")]
 mod size_asserts {
     use rustc_data_structures::static_assert_size;
--- compiler/rustc_errors/src/lib.rs
+++ compiler/rustc_errors/src/lib.rs
@@ -102,5 +102,6 @@
 // `PResult` is used a lot. Make sure it doesn't unintentionally get bigger.
 #[cfg(target_pointer_width = "64")]
 rustc_data_structures::static_assert_size!(PResult<'_, ()>, 24);
+#[cfg(not(rust_compiler="mrustc"))]
 #[cfg(target_pointer_width = "64")]
 rustc_data_structures::static_assert_size!(PResult<'_, bool>, 24);
--- compiler/rustc_parse/src/parser/mod.rs
+++ compiler/rustc_parse/src/parser/mod.rs
@@ -229,3 +229,4 @@
 // though, because `TokenTypeSet(u128)` alignment varies on others, changing the total size.
+#[cfg(not(rust_compiler="mrustc"))]
 #[cfg(all(target_pointer_width = "64", any(target_arch = "aarch64", target_arch = "x86_64")))]
 rustc_data_structures::static_assert_size!(Parser<'_>, 288);
--- compiler/rustc_middle/src/ty/sty.rs
+++ compiler/rustc_middle/src/ty/sty.rs
@@ -2026,4 +2026,4 @@
 impl<'tcx> rustc_type_ir::inherent::Tys<TyCtxt<'tcx>> for &'tcx ty::List<Ty<'tcx>> {
     fn inputs(self) -> &'tcx [Ty<'tcx>] {
-        self.split_last().unwrap().1
+        (**self).split_last().unwrap().1
     }
--- compiler/rustc_middle/src/ty/generic_args.rs
+++ compiler/rustc_middle/src/ty/generic_args.rs
@@ -703,7 +703,7 @@
 impl<'tcx, T: TypeVisitable<TyCtxt<'tcx>>> TypeVisitable<TyCtxt<'tcx>> for &'tcx ty::List<T> {
     #[inline]
     fn visit_with<V: TypeVisitor<TyCtxt<'tcx>>>(&self, visitor: &mut V) -> V::Result {
-        walk_visitable_list!(visitor, self.iter());
+        walk_visitable_list!(visitor, (***self).iter());
         V::Result::output()
     }
 }
--- vendor/security-framework-3.2.0/src/trust.rs
+++ vendor/security-framework-3.2.0/src/trust.rs
@@ -181,10 +181,13 @@
 
     /// Attaches Online Certificate Status Protocol (OSCP) response data
     /// to this trust object.
-    pub fn set_trust_ocsp_response<I: Iterator<Item = impl AsRef<[u8]>>>(
+    pub fn set_trust_ocsp_response<I: Iterator>(
         &mut self,
         ocsp_response: I,
-    ) -> Result<()> {
+    ) -> Result<()>
+    where
+        I::Item: AsRef<[u8]>,
+    {
         let response: Vec<CFData> = ocsp_response
             .into_iter()
             .map(|bytes| CFData::from_buffer(bytes.as_ref()))
@@ -197,10 +200,13 @@
 
     /// Attaches signed certificate timestamp data to this trust object.
     #[cfg(any(feature = "OSX_10_14", target_os = "ios", target_os = "tvos", target_os = "watchos", target_os = "visionos"))]
-    pub fn set_signed_certificate_timestamps<I: Iterator<Item = impl AsRef<[u8]>>>(
+    pub fn set_signed_certificate_timestamps<I: Iterator>(
         &mut self,
         scts: I,
-    ) -> Result<()> {
+    ) -> Result<()>
+    where
+        I::Item: AsRef<[u8]>,
+    {
         let scts: Vec<CFData> = scts
             .into_iter()
             .map(|bytes| CFData::from_buffer(bytes.as_ref()))
            --- vendor/security-framework-3.2.0/src/trust_settings.rs
+++ vendor/security-framework-3.2.0/src/trust_settings.rs
@@ -156,7 +156,7 @@
             let mut array_ptr: CFArrayRef = ptr::null_mut();
             let cert_ptr = cert.as_CFTypeRef() as *mut _;
             cvt(SecTrustSettingsCopyTrustSettings(cert_ptr, self.domain.into(), &mut array_ptr))?;
-            CFArray::<CFDictionary>::wrap_under_create_rule(array_ptr)
+            CFArray::<CFDictionary<CFTypeRef, CFTypeRef>>::wrap_under_create_rule(array_ptr)
         };
 
         for settings in trust_settings.iter() {
